mindmeld/components/entity_recognizer.py

```diff
@@ -119,16 +128,20 @@ def fit(self,
         # Load labeled data
         examples, labels = self._get_examples_and_labels(queries)
 
-        # Build entity types set
-        self.entity_types = set()
-        for label in labels:
-            for entity in label:
-                self.entity_types.add(entity.entity.type)
+        if examples:
+            # Build entity types set
+            self.entity_types = set()
```

Comment:
`self.entity_types = {entity.entity.type for label in labels for entity in label}`